gwpy/segments/segments.py

```diff
@@ -144,6 +145,24 @@ def coalesce(self):
         return self
     coalesce.__doc__ = segmentlist.coalesce.__doc__
 
+    def to_table(self):
+        """Convert this `SegmentList` to a `~astropy.table.Table`
+
+        The resulting `Table` has four columns: `index`, `start`, `end`, and
+        `duration`, corresponding to the zero-counted list index, GPS start
+        and end times, and total duration in seconds, respectively.
+
+        This method exists mainly to provide a way to write `SegmentList`
+        objects in comma-separated value (CSV) format, via the
+        :meth:`~astropy.table.Table.write` method.
+        """
+        return Table([
+            list(range(len(self))),
+            [seg[0] for seg in self],
+            [seg[1] for seg in self],
+            [abs(seg) for seg in self],
+        ], names=('index', 'start', 'end', 'duration'))
```

Comment:
Looking at this again, does it make more sense to only unpack the segmentlist once:

```python
        return Table(
            list(zip(*((i, s[0], s[1], abs(s)) for i, s in enumerate(a)))),
            names=('index', 'start', 'end', 'duration'),
        )

Reply:
@duncanmmacleod, this speeds up the unit test by another factor of 2 (and takes up fewer lines), so I went ahead and pushed it with the last commit.